src/main.ts

```diff
@@ -156,10 +156,19 @@ export function activate(context: vscode.ExtensionContext): ReturnType<typeof ge
             if (configuration.get('latex.autoBuild.run') as string === BuildEvents.onSave) {
                 if (extension.builder.disableBuildAfterSave) {
                     extension.logger.addLogMessage('Auto Build Run is temporarily disabled during a second.')
-                    return
+                } else {
+                    extension.logger.addLogMessage(`Auto build started on saving file: ${e.fileName}`)
+                    void extension.manager.buildOnSave(e.fileName)
                 }
-                extension.logger.addLogMessage(`Auto build started on saving file: ${e.fileName}`)
-                void extension.manager.buildOnSave(e.fileName)
+            }
+            if (configuration.get('texcount.autorun') as string === 'onSave') {
+                if (extension.counter.disableCountAfterSave) {
+                    extension.logger.addLogMessage('Auto texcount is temporarily disabled during a second.')
+                } else {
+                    extension.logger.addLogMessage(`Auto texcount started on saving file: ${e.fileName}`)
+                    void extension.counter.countOnSave()
+                }
+
```

Comment:
We had not better put everything into `src/main.ts`.

How about calling `vscode.workspace.onDidSaveTextDocument` in the constructor of `Counter`.

Reply:
Thanks for pointing out `vscode.workspace.onDidChangeConfiguration`. It can be quite handy.

I prefer to have a single call to `vscode.workspace.onDidSaveTextDocument` in the whole extension to ease maintenance. In particular, a single call to `vscode.workspace.onDidSaveTextDocument` enables us to finely monitor the execution of the different actions. Yet, we could move all the tests and configuration retrieval from

https://github.com/James-Yu/LaTeX-Workshop/blob/bd2454690afd1e346ace2db71e2815105c7ce35a/src/main.ts#L164-L172

to `extension.counter.countOnSave` in order to keep the callback concise. Would that be fine with you?

Reply:
Yes, it is fine.

Reply:
Great. I guess we should do with the same with https://github.com/James-Yu/LaTeX-Workshop/blob/7a4d03de17b0ead6784f346630c3b8bebaa0c7b0/src/main.ts#L156-L163

I will handle this later in a separate PR.

Reply:
The whole triggering logic is now implemented in `Counter` and we rely on `onDidChangeConfiguration` to read the related configuration variables.